File: provider/src/main/java/com/raygun/raygun4android/utils/RaygunFileFilter.java
package com.raygun.raygun4android.utils;

import com.raygun.raygun4android.RaygunSettings;

import java.io.File;
import java.io.FileFilter;
import java.util.Locale;

public class RaygunFileFilter implements FileFilter {
    @Override
    public boolean accept(File pathname) {

        String extension = "." + RaygunSettings.DEFAULT_FILE_EXTENSION;
        return pathname.getName().toLowerCase(Locale.ROOT).endsWith(extension);

    }
}

File: provider/src/main/java/com/raygun/raygun4android/utils/RaygunFileFilter.kt
package com.raygun.raygun4android.utils

import com.raygun.raygun4android.RaygunSettings
import java.io.File
import java.io.FileFilter

class RaygunFileFilter : FileFilter {
    override fun accept(pathname: File): Boolean {
        val extension = "." + RaygunSettings.DEFAULT_FILE_EXTENSION
        return pathname.name.lowercase().endsWith(extension)
    }
}
